lib/controllers/ephemeral/pin_entry_controller.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+import 'package:flutter/cupertino.dart';
+import 'package:get/get.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:pin_code_fields/pin_code_fields.dart';
+import 'package:pispapp/controllers/ephemeral/local_auth_controller.dart';
+import 'package:pispapp/routes/app_pages.dart';
+
+class PINEntryController extends GetxController {
+  // Constructor to fetch PIN automatically
+  PINEntryController() {
+    _fetchPIN();
+  }
+
+  // Constants
+  static const String key = 'USER_PIN';
+  static const Duration timeoutDuration = Duration(milliseconds: 1000);
+  static const int PINlength = 6;
+
+  // Storage
+  final FlutterSecureStorage _storage = const FlutterSecureStorage();
+
+  // Fields to be populated
+  String _correctPIN = '';
+  bool _userSetPIN = false;
+
+  // Returns whether or not the user set a PIN
+  bool get userSetPIN => _userSetPIN;
+
+  void onFallbackVerificationNeeded() {
+    Get.toNamed<dynamic>(Routes.PIN_ENTRY);
+  }
+
+  // Called after correct PIN has been populated
+  bool authenticate(String pin) => pin == _correctPIN;
+
+  // Authentication process
+  void onPINEntered(String pin, TextEditingController tc, StreamController<ErrorAnimationType> ec) {
+    if(!_userSetPIN) { return; } // Do nothing if there is no previous set PIN
```

Comment:
what if the pin is not set because it has not been fetched?

Also, I think it would be better to throw an error here?

Reply:
I think the way this works is slightly confusing. When there's no set PIN, this function should do nothing since this is called when the user fills in the fields of the PIN. When there's no set PIN, a button appears below that allows the user to save the PIN so we shouldn't be trying to verify the PIN at all. 

I moved this logic out into the callback instead of leaving it in the controller

Reply:
I think we should split the function for when user wants to save a pin and when we need to do verification